apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java

```diff
@@ -884,18 +952,99 @@ private void addC4DicIdentifier(Coverage coverage, Beneficiary beneficiary) {
         TransformerUtilsV2.findOrCreateContainedOrganization(
             coverage, TransformerUtilsV2.PROVIDER_ORG_ID, Profile.C4DIC);
 
-    Identifier identifier =
-        new Identifier()
-            .setType(
-                TransformerUtilsV2.createCodeableConcept(
-                    TransformerConstants.CODING_SYSTEM_HL7_IDENTIFIER_TYPE,
-                    null,
-                    TransformerConstants.PATIENT_MB_ID_DISPLAY,
-                    "MB"))
-            .setValue((String.valueOf(beneficiary.getBeneficiaryId())))
-            .setSystem(TransformerConstants.CODING_BBAPI_BENE_ID)
-            .setAssigner(new Reference(organization));
-    coverage.addIdentifier(identifier);
+    if (beneficiary.getMedicareBeneficiaryId().isPresent()) {
+      Identifier identifier =
+          new Identifier()
+              .setType(
+                  TransformerUtilsV2.createCodeableConcept(
+                      TransformerConstants.CODING_SYSTEM_HL7_IDENTIFIER_TYPE,
+                      null,
+                      TransformerConstants.PATIENT_MB_ID_DISPLAY,
+                      "MB"))
+              .setValue(beneficiary.getMedicareBeneficiaryId().get())
+              .setSystem(TransformerConstants.CODING_BBAPI_MEDICARE_BENEFICIARY_ID_UNHASHED)
+              .setAssigner(new Reference(organization));
+      coverage.addIdentifier(identifier);
+    }
+  }
+
+  /**
+   * Denotes which parts the beneficiary is subscribed to on the provided {@link Coverage} resource.
+   *
+   * @param beneficiary the value for {@link Beneficiary}
+   * @param coverage The {@link Coverage} to Coverage details
+   */
+  private void addSubscribedParts(Beneficiary beneficiary, Coverage coverage) {
+    if (beneficiary.getPartACoverageStartDate().isPresent()) {
+      createCoverageClass(
+          coverage,
+          CoverageClass.PLAN,
+          TransformerConstants.COVERAGE_PLAN_PART_A,
+          Optional.empty());
+    }
+    if (beneficiary.getPartBCoverageStartDate().isPresent()) {
+      createCoverageClass(
+          coverage,
+          CoverageClass.PLAN,
+          TransformerConstants.COVERAGE_PLAN_PART_B,
+          Optional.empty());
+    }
+    Optional<Instant> lastUpdatedOpt = beneficiary.getLastUpdated();
+    if (lastUpdatedOpt.isPresent()) {
```

Comment:
Is it possible for `last_updated` to be null if it's the first time the bene is loaded in? If so, we may want to add a fallback here to check the current month if we don't have a `last_updated` time.